civis/civis.py

```diff
@@ -13,6 +13,45 @@
 
 
 def find(object_list, filter_func=None, **kwargs):
+    """Return the objects from ``object_list`` that satisfy the filters.
+
+    Parameters
+    ----------
+    object_list : iterable
+        An iterable of arbitrary objects, particularly dicts and objects with
+        attributes that can be targeted by the filters in ``kwargs``.
+    filter_func : callable, optional
+        A one-argument function. If specified, ``kwargs`` are ignored.
+        An ``object`` from the input iterable is kept in the returned list
+        if and only if ``bool(filter_func(object))`` is ``True``.
+    **kwargs
+        Key-value pairs for more fine-grained filtering; they cannot be used
+        in conjunction with ``filter_func``. All keys must be strings.
+        An ``object`` from the input iterable is kept in the returned
+        list, unless one of the following conditions is met:
+        - ``key`` is not an attribute of ``object``
+        - ``value`` is a one-argument function and
+          ``bool(value(object[key]))`` is ``False``
+        - ``value`` is ``False``
```

Comment:
Wow, this is confusing, and even seems buggy to me. This means that if `obj.spam = False`, and you run `civis.find([obj], spam=False)`, you get back an empty list, not the object for which `spam` is `False`.

If you provide a boolean `value`, you're nominally filtering on whether or not an object contains the `key` as an attribute. Except that's very different than what happens if you provide a non-boolean `value` (we check for equality). And it doesn't even work, because if you provide a key which isn't an attribute and the value `False`, then the object still isn't included. It looks like providing a `False` as the value to any kwarg will always result in an empty list being returned.